File: 【栈】2023C-密码输入检测.cpp
/*
 * @hw id=2023C lang=cpp
 *
 * 密码输入检测
 */

// @hw code=start
#include<iostream>
#include<vector>
using namespace std;

int main() {
    string s;
    getline(cin, s);

    vector<char> stack;
    for (char c : s) {
        if(c == '<' && stack.size() >= 1) {
            stack.pop_back();
        } else if(c!='<') {
            stack.push_back(c);
        }
    }

    int flag = 0;
    for (char c : stack) {
        if(isupper(c)) {
            flag |= 0b0001;
        } else if(islower(c)) {
            flag |= 0b0010;
        } else if(isdigit(c)) {
            flag |= 0b0100;
        } else if(!isspace(c) && !isalnum(c)) {
            flag |= 0b1000;
        }
    }

    string ans;
    for(char c : stack) {
        ans += c;
    }
    
    if(flag == 0b1111 && stack.size() >= 8) {
        cout << ans << ",true" << endl;
    } else {
        cout << ans << ",false" << endl;
    }
    return 0;
}
// @hw code=end

File: 【栈】2023C-密码输入检测.java
/*
 * @hw id=2023C lang=java
 *
 * 密码输入检测
 */

// @hw code=start
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        String s = in.nextLine();
        List<Character> stack = new ArrayList<>();
        for (char c : s.toCharArray()) {
            if(c == '<' && stack.size() >= 1) {
                stack.remove(stack.size() - 1);
            } else if(c != '<') {
                stack.add(c);
            }
        }
        
        int flag = 0;
        for(char c : stack) {
            if(Character.isUpperCase(c)) {
                flag |= 0b0001;
            } else if (Character.isLowerCase(c)) {
                flag |= 0b0010;
            } else if (Character.isDigit(c)) {
                flag |= 0b0100;
            } else if (!Character.isLetterOrDigit(c) && !Character.isWhitespace(c)) {
                flag |= 0b1000;
            }
        }
        StringBuilder sb = new StringBuilder();

        for(int i=0;i<stack.size();i++) {
            sb.append(stack.get(i));
        }

        if(flag==0b1111 && stack.size() >= 8) {
            System.out.println(sb.toString() + ",true");
        } else {
            System.out.println(sb.toString() + ",false");
        }
    }
}
// @hw code=end